AfrikaBurn/MapViewController.swift

```diff
@@ -0,0 +1,339 @@
+//
+//  MapViewController.swift
+//  AfrikaBurn
+//
+//  Created by Daniel Galasko on 2017/03/04.
+//  Copyright © 2017 AfrikaBurn. All rights reserved.
+//
+
+import UIKit
+import MapKit
+import RealmSwift
+
+struct Resources {
+    static let mapOverlayImage: UIImage = {
+        let path = Bundle.main.path(forResource: "AB2017_Layout_RoadsLabel_20170308", ofType: "png")!
+        return UIImage(contentsOfFile: path)!
+    }()
+}
+
+struct BurnMap {
+    static let boundingMapRect: MKMapRect = MKMapRect(origin: MKMapPoint(x: 148937508.42330855, y: 159711548.54660633), size: MKMapSize(width: 9388.6817751824855, height: 7286.825931340456))
+}
+
+class MapViewController: UIViewController {
+    
+    var selectedElement : AfrikaBurnElement? = nil
+    @IBOutlet weak var mapView: MKMapView!
+    let persistentStore = PersistentStore()
+    let locationManager = CLLocationManager()
+    lazy var allElements: Results<AfrikaBurnElement> = self.persistentStore.elements()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        locationManager.requestWhenInUseAuthorization()
+        mapView.showsUserLocation = true
+        mapView.delegate = self
+        
+        attachLongPress()
+        
+        loadCustomLocations()
+        loadElements()
+        
+        navigationItem.title = "Map"
+        navigationController?.navigationBar.titleTextAttributes =  [NSForegroundColorAttributeName: UIColor.afrikaBurnTintColor]
+        navigationController?.navigationBar.barTintColor = UIColor.afrikaBurnBgColor
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+    }
+    
+    func attachLongPress(){
+        let gesture = UILongPressGestureRecognizer(target: self, action: #selector(MapViewController.longPressOnMap(gestureRecognizer:)))
+        gesture.minimumPressDuration = 2.0
+        
+        mapView.addGestureRecognizer(gesture)
+        
+        
+    }
+    
+    func longPressOnMap(gestureRecognizer:UIGestureRecognizer){
+        
+        if (gestureRecognizer.state != .began) {
```

Comment:
I guard here might read better in that `guard state == .began else` reads clearly whereas the inverted if I still need to process a bit to understand whats happening after